src/firmware/shared/circular_buffer.c

```diff
@@ -0,0 +1,63 @@
+#include "firmware/shared/circular_buffer.h"
+
+#include <assert.h>
+
+struct CircularBuffer
+{
+    int head;
+    int tail;
+    int size;
```

Comment:
Perhaps `max_size` might be a bit more descriptive, since it can technically be smaller then this (for example, if we haven't pushed enough elements to it to fill it up)

Reply:
Changed to `max_size`